src/java/org/apache/cassandra/db/virtual/AbstractLoggerVirtualTable.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.virtual;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ch.qos.logback.classic.spi.LoggingEvent;
+import org.apache.cassandra.schema.TableMetadata;
+
+/**
+ * This table is inherently limited on number of rows it can hold.
+ *
+ * @param <U> type parameter saying what object is stored in internal bounded list for query purposes
+ */
+public abstract class AbstractLoggerVirtualTable<U> extends AbstractMutableVirtualTable
+{
+    private static final Logger logger = LoggerFactory.getLogger(AbstractLoggerVirtualTable.class);
+
+    protected final List<U> buffer;
```

Comment:
probably we need to add a comment here that all access to this field must be protected by synchronising on it..

Reply:
this is backed by `Collections.synchronizedList(new BoundedLinkedList<>(size));`

Reply:
this is already backed by synchronized list but sure ...

Reply:
yes, but as I see there are tricky places in the code (handled correctly) when we have to use extra synchronisation, for example listIterator